dhall-kubernetes-generator/src/Main.hs

```diff
@@ -41,21 +52,71 @@ echo = Turtle.printf (Turtle.s Turtle.% "\n")
 echoStr :: Turtle.MonadIO m => String -> m ()
 echoStr = echo . Text.pack
 
+errorOnDuplicateHandler :: DuplicateHandler
+errorOnDuplicateHandler (kind, names) = error $ "Got more than one key for "++ show kind ++"! See:\n" <> show names
+
+skipDuplicatesHandler :: DuplicateHandler
+skipDuplicatesHandler = const Nothing
+
+parseImport :: String -> Expr -> Dhall.Parser.Parser Dhall.Import
+parseImport _ (Dhall.Note _ (Dhall.Embed l)) = pure l
+parseImport prefix e = fail $ "Expected a Dhall import for " <> prefix <> " not:\n" <> show e
+
+parsePrefixMap :: Options.Applicative.ReadM PrefixMap
+parsePrefixMap =
+  Options.Applicative.eitherReader $ \s ->
+    bimap errorBundlePretty Data.Map.fromList $ result (pack s)
+  where
+    parser = do
+      prefix <- some (alphaNumChar <|> char '.')
+      char '='
+      e <- Dhall.Parser.expr
+      imp <- parseImport prefix e
+      optional $ char ','
+      return (pack prefix, imp)
+    result = parse (some (Dhall.Parser.unParser parser)) "MAPPING"
+
+parseOptions :: Options.Applicative.Parser Options
+parseOptions = Options <$> parseSkip <*> parsePrefixMap' <*> fileArg
+  where
+    parseSkip =
+      Options.Applicative.switch
+        (  Options.Applicative.long "skipDuplicates"
+        <> Options.Applicative.help "Skip types with the same name when aggregating types"
+        )
+    parsePrefixMap' =
+      Options.Applicative.option parsePrefixMap
+        (  Options.Applicative.long "prefixMap"
+        <> Options.Applicative.help "Specify prefix mappings as 'prefix1=importBase1,prefix2=importBase2,...'"
+        <> Options.Applicative.metavar "MAPPING"
+        )
+    fileArg = Options.Applicative.strArgument
+            (  Options.Applicative.help "The swagger file to read"
+            <> Options.Applicative.metavar "FILE"
+            )
+
+-- | `ParserInfo` for the `Options` type
+parserInfoOptions :: Options.Applicative.ParserInfo Options
+parserInfoOptions =
+    Options.Applicative.info
+        (Options.Applicative.helper <*> parseOptions)
+        (   Options.Applicative.progDesc "Swagger to Dhall generator"
+        <>  Options.Applicative.fullDesc
+        )
 
 main :: IO ()
 main = do
+  options <- Options.Applicative.execParser parserInfoOptions
```

Comment:
If you enable the `RecordWildCards` language extension then you can do:

```suggestion
  Options{..} <- Options.Applicative.execParser parserInfoOptions
```

... and then instead of `skipDuplicates options`/`prefixMap options` you can just write `skipDuplicates`/`prefixMap`